### PR TITLE
feat(courses):[SKP-481] call filter courses action

### PR DESCRIPTION
**Pull request by MatuzasM**

### CODE REVIEW

File: libs/data-access/courses/src/lib/saga.ts [hunk 1]
```diff
@@ -263,6 +270,25 @@ export function* removeStudentFromCourse(actions) {
   }
 }
 
+export function* filterCourses(actions) {
+  try {
+    const path = `/api/${PrivateRoutesPath.COURSES}/filter?`;
+    const result = yield call(getByQueryParams, path, actions.payload);
+    if (result) {
```
Comment (doncevas) [inline libs/data-access/courses/src/lib/saga.ts:277]: what should happen if result return null?

File: libs/shared/filter/src/lib/shared-filter.tsx [hunk 2]
```diff
@@ -8,64 +8,210 @@ import {
...
-      <Accordion elevation={2} square={true}>
-        <AccordionSummary
-          expandIcon={<ExpandMoreIcon />}
-          aria-controls="panel1a-content"
-          id="panel1a-header"
+    <div className="filter-form">
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <div
+          className="rating"
+          style={{
```
Comment (doncevas) [inline libs/shared/filter/src/lib/shared-filter.tsx:52]: don't use inline style, we have style file, same below

**Review by doncevas — CHANGES_REQUESTED** [review]

File: libs/data-access/courses/src/lib/saga.ts (hunk 1 above)
Comment (MatuzasM) [reply libs/data-access/courses/src/lib/saga.ts:277]: it should return all courses?

File: libs/features/courses/src/lib/features-courses.tsx [hunk 3]
```diff
@@ -117,88 +67,89 @@ export const FeaturesCourses = () => {
...
+          </div>
+        </Grid>
+        <Grid
+          item
+          xs={6}
+          md={isFilterActive === true ? 9 : 11}
+          className="course-cards"
+        >
+          <div>
+            {coursesState.length &&
```
Comment (doncevas) [inline libs/features/courses/src/lib/features-courses.tsx:113]: coursesState?.length

**Review by doncevas — CHANGES_REQUESTED** [review]

**Review by doncevas — APPROVED** [review]

**Pull request merged**